monai/apps/reconstruction/mri_transforms.py

```diff
@@ -0,0 +1,301 @@
+        probability that 8-fold acceleration with 4% center fraction is selected.
+    """
+
+    def __call__(self, spatial_size: Sequence[int], seed: Optional[int] = None) -> Tensor:
+        """
+        Args:
+            spatial_size: The shape of the mask to be created. The shape should have
+                at least 3 dimensions (H,W,2). Samples are drawn along the second last dimension (W).
+            seed: Seed for the random number generator. Setting the seed
+                ensures the same mask is generated each time for the same shape.
+
+        Returns:
+            final_mask which is the under-sampling mask.
+        """
+        if len(spatial_size) < 3:
+            raise ValueError("Shape should have 3 or more dimensions")
+
+        self.rng.seed(seed)
+        center_fraction, acceleration = self.choose_acceleration()
+        num_cols = spatial_size[-2]
+        num_low_freqs = int(round(num_cols * center_fraction))
+
+        # Create the mask
+        mask = np.zeros(num_cols, dtype=np.float32)
+        pad = (num_cols - num_low_freqs + 1) // 2
+        mask[pad : pad + num_low_freqs] = True
+
+        # Determine acceleration rate by adjusting for the number of low frequencies
+        adjusted_accel = (acceleration * (num_low_freqs - num_cols)) / (num_low_freqs * acceleration - num_cols)
+        offset = self.rng.randint(0, round(adjusted_accel))
+
+        accel_samples = np.arange(offset, num_cols - 1, adjusted_accel)
+        accel_samples = np.around(accel_samples).astype(np.uint)
+        mask[accel_samples] = True
+
+        # Reshape the mask
+        mask_shape = [1 for _ in spatial_size]
+        mask_shape[-2] = num_cols
+        final_mask: Tensor = convert_to_tensor(mask.reshape(*mask_shape).astype(np.float32))
+
+        return final_mask
+
+
+def apply_mask(
+    data: Tensor, mask_func: MaskFunc, mask: Optional[Tensor] = None, seed: Optional[int] = None
+) -> Sequence[Tensor]:
+    """
+    Subsample the given k-space by multiplying with a mask.
+
+    Args:
+        data: The input k-space data. This should have at least 3 dimensions (...,H,W,2), where
+            dimensions -3 and -2 are the spatial dimensions, and the final dimension has size
+            2 (for complex values).
+        mask_func: A function that takes a shape and a random
+            number seed and returns a mask.
+        mask: the undersampling mask. when this is provided,
+            mask_func will be ignored
+        seed: Seed for the random number generator.
+
+    Returns:
+        tuple containing
+            (1) masked data (torch.Tensor): Subsampled k-space data
+            (2) mask (torch.Tensor): The generated mask
+    """
+    if mask is None:
+        shape = np.array(data.shape)
+        shape[:-3] = 1
+        mask = mask_func(shape.tolist(), seed)
+    return data * mask, mask
+
+
+def create_mask_for_mask_type(
+    mask_type_str: str, center_fractions: Sequence[float], accelerations: Sequence[float]
```

Comment:
`mask_type_str` could be `mask_type` with the actual type passed rather than its name (as you've done with `mask_func` in `apply_mask`).